src/mysql-util/src/schemas.rs

```diff
@@ -7,52 +7,181 @@
+    schemas: &Option<Vec<String>>,
 ) -> Result<Vec<MySqlTableDesc>, MySqlError> {
-    // Get all tables of type 'Base Table' in schema
-    let table_q = "SELECT table_name, table_schema
-                   FROM information_schema.tables
-                   WHERE table_type = 'BASE TABLE'
-                   AND table_schema IN (?)";
-    let table_rows: Vec<(String, String)> = conn.exec(table_q, schemas).await?;
+    let table_rows: Vec<(String, String)> = if let Some(schemas) = schemas {
+        // Get all tables of type 'Base Table' in specified schemas
+        assert!(!schemas.is_empty());
+        let table_q = format!(
+            "SELECT table_name, table_schema
+                       FROM information_schema.tables
+                       WHERE table_type = 'BASE TABLE'
+                       AND table_schema IN ({})",
+            schemas.iter().map(|_| "?").join(", ")
+        );
+        conn.exec(table_q, schemas).await?
+    } else {
+        // Get all tables in non-system schemas.
+        // TODO(roshan): Many users create user-defined tables in the `mysql` system schema, since mysql doesn't
+        // prevent this. We may want to consider adding a warning for this in the docs, since that schema
+        // contains dozens of built-in system tables that we need to filter out.
+        let table_q = "SELECT table_name, table_schema
+                       FROM information_schema.tables
+                       WHERE table_type = 'BASE TABLE'
+                       AND table_schema NOT IN ('information_schema', 'performance_schema', 'mysql', 'sys')";
+        conn.exec(table_q, ()).await?
+    };
 
     let mut tables = vec![];
     for (table_name, schema_name) in table_rows {
-        let column_q = "SELECT column_name, data_type, is_nullable, ordinal_position
-                        FROM information_schema.columns
-                        WHERE table_name = ? AND table_schema = ?";
+        let column_q = format!(
+            "SELECT {}
+             FROM information_schema.columns
+             WHERE table_name = ? AND table_schema = ?
+             ORDER BY ordinal_position ASC",
+            INFO_SCHEMA_COLS.keys().join(", ")
+        );
         let column_rows = conn
-            .exec::<(String, String, String, u16), _, _>(column_q, (&table_name, &schema_name))
+            .exec::<Row, _, _>(column_q, (&table_name, &schema_name))
             .await?;
 
         let mut columns = Vec::with_capacity(column_rows.len());
-        for (name, data_type, is_nullable, ordinal_position) in column_rows {
-            let data_type = match data_type.as_str() {
-                "int" => MySqlDataType::Int,
-                "varchar" => MySqlDataType::Varchar(usize::MAX),
-                "char" => MySqlDataType::Varchar(usize::MAX),
-                _ => return Err(MySqlError::UnsupportedDataType(data_type)),
+        for mut row in column_rows {
+            let name = row.take(INFO_SCHEMA_COLS["column_name"]).unwrap();
```

Comment:
Ah yes, I saw that too and originally implemented all of this using `row.take("column_name").unwrap()`, etc but unfortunately hit runtime errors -- `take()` would always return `None` when using the `&str` index-type. Based on your other comment I plan to switch to specifying the types directly -- I thought this might be more ergonomic but it doesn't look that way anymore.

Separately I do want to figure out why the `Row.take(&str)` method isn't working correctly -- I may file an issue with `mysql-async`